qiskit_dynamics/signals/signals.py

```diff
@@ -895,9 +858,9 @@ def signal_add(sig1: Signal, sig2: Signal) -> SignalSum:
             and sig1.start_time == sig2.start_time
             and sig1.duration == sig2.duration
         ):
-            samples = np.append(sig1.samples, sig2.samples, axis=1)
-            carrier_freq = np.append(sig1.carrier_freq, sig2.carrier_freq)
-            phase = np.append(sig1.phase, sig2.phase)
+            samples = unp.append(sig1.samples, sig2.samples, axis=1)
+            carrier_freq = unp.append(sig1.carrier_freq, sig2.carrier_freq)
+            phase = unp.append(sig1.phase, sig2.phase)
```

Comment:
I suspect these lines may be prone to the dispatching errors we had to deal with in other places. Should these be changed to use `_numpy_multi_dispatch`?

Reply:
I forgot to change it to `_numpy_multi_dispatch`.
I fixed at https://github.com/Qiskit-Extensions/qiskit-dynamics/pull/269/commits/2e1436e838dec1acbb5dd478b39a773d353741b4.